src/util/byteswap.c

```diff
@@ -23,26 +23,28 @@
 #include <stdint.h>
 #include "byteswap.h"
 
-//! Byte swap unsigned short
 uint16_t swap_uint16( uint16_t val )
 {
     return (val << 8) | (val >> 8 );
 }
 
-//! Byte swap short
 int16_t swap_int16( int16_t val )
 {
     return (val << 8) | ((val >> 8) & 0xFF);
 }
 
-//! Byte swap unsigned int
+uint32_t swap_uint24( int32_t val )
+{
+    val = ((val & 0xFF0000) >> 16) | (val & 0x00FF00) | ((val & 0x0000FF) << 16);
```

Comment:
I collapsed the two statements. I didn't make the parameter const since it's passing by value anyway.